text/0136-mobile-tracing-without-performance-v-2.md

```diff
@@ -0,0 +1,74 @@
+- Start Date: 2024-06-04
+- RFC Type: decision
+- RFC PR: https://github.com/getsentry/rfcs/pull/136
+- RFC Status: draft
+
+# Summary
+
+This RFC aims to find a strategy to update the  so traces don’t reference hundreds of unrelated
+events.
+
+# Motivation
+
+On mobile, traces can have hundreds of unrelated events caused by the possibly never-changing
+`traceId` required for tracing without performance. This occurs mostly when users don’t have performance
+enabled.
+
+# Background
+
+In the summer of 2023, all mobile SDKs implemented [Tracing without performance](https://www.notion.so/Tracing-without-performance-efab307eb7f64e71a04f09dc72722530?pvs=21),
+see also [team-sdks GH issue](https://github.com/getsentry/team-sdks/issues/5).
+The goal of this endeavor was to
+
+> always have access to a trace and span ID, add a new internal `PropagationContext` property to the
+> scope, an object holding a `traceId` and `spanId`
+
+On mobile, most users interact purely with the static API, which holds a reference to a global
+Hub and Scope. Therefore, mobile SDKs create a `PropagationContext` with `traceId` and `spanId`
+during initialization, and these usually persist for the entire lifetime of the app. Mobile
+SDKs prefer the `traceID` of transactions bound to the scope over the `PropagationContext`. So
+when performance is disabled, or no transition is bound to the scope, mobile SDKs use the same
+`traceId` and `spanId` for all captured events. This can lead to traces with hundreds of
+unrelated events confusing users. JS addressed this recently by updating the `PropagationContext`
+based on routes, see [Ensure browser traceId lifetime works as expected](https://github.com/getsentry/sentry-javascript/issues/11599).
+
+# Options Considered
+
+## Option 1: Update `PropagationContext` based on screens <a name="option-1"></a>
```

Comment:
I'm wondering if we should focus on navigation instead and have screens as a fallback. At least on Android there are a few widely used navigation frameworks (like https://developer.android.com/guide/navigation) which provide an URL-like route.

Reply:
If you can use routes, use routes, as described below. I updated the title.